include/deal.II/lac/solver_richardson.h

```diff
@@ -143,33 +142,17 @@ class SolverRichardson : public Solver<VectorType>
 protected:
   /**
    * Implementation of the computation of the norm of the residual.
+   * Depending on the flags given to the solver, the default
+   * implementation of this function uses either the actual
+   * residual, @p r, or the preconditioner residual, @p d.
    */
-  virtual typename VectorType::value_type criterion();
-
-  /**
-   * Residual. Temporary vector allocated through the VectorMemory object at
-   * the start of the actual solution process and deallocated at the end.
-   */
-  VectorType *Vr;
-  /**
-   * Preconditioned residual. Temporary vector allocated through the
-   * VectorMemory object at the start of the actual solution process and
-   * deallocated at the end.
-   */
-  VectorType *Vd;
+  virtual typename VectorType::value_type criterion(const VectorType &r,
+                                                    const VectorType &d) const;
```

Comment:
Do we need an entry in the changelog for this incompatibility? The function is protected...